getting_started/step_by_step/your_first_game.rst

```diff
@@ -176,15 +176,20 @@ Start by declaring the member variables this object will need:
 
     extends Area2D
 
-    export var speed = 400  # How fast the player will move (pixels/sec).
-    var screen_size  # Size of the game window.
+    # These only need to be accessed in this script, so we can make them private.
+    # Private variables in GDScript have their name starting with an underscore.
```

Comment:
GDScript doesn't have private variables per se, so this might be a bit misleading.

"Your first game" is also supposed to be simple, personally I'm not sure whether introducing users to public/private access at this stage is really useful.

Good practices are good but we should be careful with the simpler demos not to increase the overhead needlessly for beginners (that's why we don't recommend using type hints in simple demos for example).

Reply:
In our tuts we call these pseudo-private.

In the docs, I'd assume people have watched a CS intro course or have programming foundations; we'll link to some free resources as part of the getting started rewrite. Other pages already talk about programming patterns or classes, and there's not really a way around it considering Godot is an object-oriented engine, even on the user's side.

Here, the properties could well stay public. I don't know what's best.

Reply:
I agree with @akien-mga, this is introducing a whole set of concepts that are really of no help in making this program work. It only adds to the cognitive load - if you understand public/private, it's misleading, and if you don't, it's just confusing.

Reply:
I imagine the demo in godot-demo-projects was updated to be like that. So it's not you who changed the code, @cbscribe, right? I took it as you had updated the code or someone had consulted you before changing it.

Reply:
No, I haven't done any updates to the demo project. I did see there were changes to the C# version being made recently, but I didn't review them. Maybe it was slipped in during those?

Reply:
The changes were also applied to the GDScript version, yes: https://github.com/godotengine/godot-demo-projects/commit/dd11042d95d0cb996b815818d2766b35ad211e8e

Reply:
It's easy to undo the changes to the demo projects, we can change these back to public.